frontend/javascripts/oxalis/model/sagas/volumetracing_saga.js

```diff
@@ -235,6 +256,172 @@ function* copySegmentationLayer(action: CopySegmentationLayerAction): Saga<void>
+      [x - inputContextExtent, y - inputContextExtent, z],
+      activeViewport,
+    );
+    const max = Dimensions.transDim(
+      [x - inputContextExtent + inputExtent, y - inputContextExtent + inputExtent, z + 1],
+      activeViewport,
+    );
+
+    const cuboidData = await api.data.getDataFor2DBoundingBox(colorLayer.name, {
+      min,
+      max,
+    });
+
+    const tensorArray = new Float32Array(new Uint8Array(cuboidData)).map(
+      el => (el - mean) / stdDev,
+    );
+
+    return useWebworker
+      ? workerPredict(useGPU, tensorArray.buffer, inputExtent)
+      : mainThreadPredict(useGPU, tf, tensorArray.buffer, inputExtent);
+  };
+
+  function* currentPositionUpdater(): Saga<void> {
+    while (!aborted) {
+      yield* take(FlycamActions);
+      const curPosition = Dimensions.transDim(
+        Dimensions.roundCoordinate(yield* select(state => getPosition(state.flycam))),
+        activeViewport,
+      );
+      [curX, curY, curZ] = curPosition;
+    }
+  }
+
+  const getter = async (x, y, z): Promise<boolean> => {
+    if (
+      x < curX - halfViewportExtentX ||
+      y < curY - halfViewportExtentY ||
+      z < curZ ||
+      x >= curX + halfViewportExtentX ||
+      y >= curY + halfViewportExtentY
+    ) {
+      return false;
+    }
+
+    // If the current z-slice is too far ahead, wait
+    // eslint-disable-next-line no-await-in-loop
+    while (z >= curZ + NUM_PREDICT_SLICES && !aborted) await sleep(500);
+
+    const tileX = Math.floor(x / outputExtent);
+    const tileY = Math.floor(y / outputExtent);
+    const tileZ = z;
+
+    if (predictions.get([tileX, tileY, tileZ]) == null) {
+      // Do not make new predictions if the magic wand has been aborted
+      if (aborted) return false;
+      console.time("predict");
+      Toast.info(`${toastDescription}\nLabeling slice ${z}.`, toastConfig);
+      predictions.set([tileX, tileY, tileZ], await getPredictionForTile(tileX, tileY, tileZ));
+      console.timeEnd("predict");
+    }
+
+    const relX = x % outputExtent;
+    const relY = y % outputExtent;
+    const prediction = predictions.get([tileX, tileY, tileZ]);
+    if (prediction != null) return prediction[relX * outputExtent + relY] > FLOODFILL_THRESHOLD;
+    throw new Error("This should never happen, prediction was set, but geting it failed.");
+  };
+
+  const onFlood = async (floodedVoxels: Array<Vector3>) => {
+    console.time("label");
+    const voxelAddresses = floodedVoxels.map(addressUVW =>
+      Dimensions.transDim(addressUVW, activeViewport),
+    );
+    api.data.labelVoxels(voxelAddresses, activeCellId);
+    console.timeEnd("label");
+    // Do not block the main thread and allow for some interactivity after labeling a slice
+    await sleep(500);
+  };
+
+  const seed = clickPosition;
+  const seedPrediction = yield* call(getter, ...seed);
+  if (seedPrediction) {
+    // The floodfill will run until aborted
+    floodfill({ getter, seed, onFlood });
```

Comment:
This is a bit hard to read, since not using the returned promise suggests a bit that the function call would be blocking. Also, couldn't it happen that the flood fill is finished (e.g., when a cell is closed?). In that case, `aborted` would need to be set to true, right? Maybe invoke this as a saga via fork?

Reply:
What I did to address this and the next comment:
- I switched the order of the `floodfill(...)` and `yield* call(currentPositionUpdater)` lines
- made the `currentPositionUpdater` call non-blocking via fork
- made the `floodfill` call blocking via call

So the saga will run until the floodfill ends, which is either once it is aborted or once it is exhausted (no more voxels to fill). Works really well :)